modules/menu/menu.c

```diff
@@ -1012,6 +1013,39 @@ struct ua   *menu_ua_carg(struct re_printf *pf, const struct cmd_arg *carg,
 	return ua;
 }
 
+/**
+ * Decode url format
+ *
+ * @param w  Pl correspond the string is in URL format will be decoded
+ */
+void url_decode(struct pl* w)
+{
+	char *dst = (char*)w->p;
+	char *src = (char*)w->p;
+	size_t len = w->l;
+	char a, b;
+	while (*src && src < w->p+len) {
+		if ((*src == '%') &&
+				((a = src[1]) && (b = src[2])) &&
```

Comment:
src[1]) and src[2] can be out of bound here if url is malformed, right? a struct pl is not necessary null terminated. so pl length has to be checked.

Reply:
I got it. Thanks for correcting me.